core/src/cdn_url.rs

```diff
@@ -100,14 +100,14 @@ impl CdnUrl {
 impl TryFrom<CdnUrlMessage> for MaybeExpiringUrls {
     type Error = crate::Error;
     fn try_from(msg: CdnUrlMessage) -> Result<Self, Self::Error> {
-        if !matches!(msg.get_result(), StorageResolveResponse_Result::CDN) {
+        if !matches!(msg.result.unwrap(), StorageResolveResponse_Result::CDN) {
```

Comment:
librespot should remain free `unwrap`s unless the result is guaranteed. Better to fail this `try_from` instead.